tests/rptest/services/utils.py

```diff
@@ -12,6 +12,10 @@ class BadLogLines(Exception):
     def __init__(self, node_to_lines):
         self.node_to_lines = node_to_lines
 
+    @staticmethod
+    def _get_hostname(node):
+        return node.account.hostname
```

Comment:
Would one of these strategies be cleaner:

# node/pod polymorphism

```diff
diff --git a/tests/rptest/services/cloud_broker.py b/tests/rptest/services/cloud_broker.py
index b24b3a636a..0a0181ba59 100644
--- a/tests/rptest/services/cloud_broker.py
+++ b/tests/rptest/services/cloud_broker.py
@@ -33,6 +33,7 @@ class CloudBroker():
         # Backward compatibility
         # Various classes will use this var to hash and compare nodes
         self.account = self._meta
+        self.account['hostname'] = f"{self._spec['nodeName']}/{self.name}"
```

# hacky polymorphism

```diff
diff --git a/tests/rptest/services/utils.py b/tests/rptest/services/utils.py
index c1c278bee0..c88bc70f86 100644
--- a/tests/rptest/services/utils.py
+++ b/tests/rptest/services/utils.py
@@ -27,6 +27,13 @@ class BadLogLines(Exception):
     def __repr__(self):
         return self.__str__()

+    @staticmethod
+    def _get_hostname(node):
+        if isinstance(node, CloudBroker):
+            return f"{node._spec['nodeName']}/{node.name}"
+        else:
+            return node.account.hostname
+
```

Reply:
but actually for least modification I agree with the first approach that both `node` and CloudBroker can expose the a "hostname" using the same API allowing the `BadLogLines` exception to work with either type (this should type cleanly as well).

Reply:
> I feel like it can also just be solved from the "outside" by making node_to_lines a map from str account name to bad log lines.

Yeh, that __str__ method doesn't really need anything else from the node/broker object from what i can tell.

Reply:
First approach is better and simpler. I guess I really froze on the idea of not touching any other module that I am working with. Will update.

Reply:
>  I guess I really froze on the idea of not touching any other module that I am working with

I would really encourage you to feel free to reach into other modules and modify them to make the final result as clean as possible, versus trying to stick to one module. I.e., the whole of rptest can be the scope for these kinds of changes.